src/hooks/v3/V3ContractLoader.ts

```diff
@@ -0,0 +1,46 @@
+import { useWallet } from 'hooks/Wallet'
+import { V3ContractName, V3Contracts } from 'models/v3/contracts'
+import { useEffect, useState } from 'react'
+
+import { loadContract } from 'utils/contracts/v3/loadContract'
+
+import { readNetwork } from 'constants/networks'
+import { readProvider } from 'constants/readProvider'
+
+export function useV3ContractLoader() {
+  const { signer } = useWallet()
+  const [contracts, setContracts] = useState<V3Contracts>()
+
+  useEffect(() => {
+    async function loadContracts() {
+      try {
+        const network = readNetwork.name
+
+        // Contracts can be used read-only without a signer, but require a signer to create transactions.
+        const signerOrProvider = signer ?? readProvider
+
+        const contractLoaders = await Promise.all(
+          Object.values(V3ContractName).map(contractName =>
+            loadContract(contractName, network, signerOrProvider),
+          ),
+        )
```

Comment:
This File is mostly a clone of V2 file, except for this line.

We can probably abstract this loadContracts hook further for better re-use/DRY